src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -418,6 +419,7 @@ pub struct CreateSinkStatement<T: AstInfo> {
     pub with_options: Vec<SqlOption<T>>,
     pub format: Option<Format<T>>,
     pub envelope: Option<Envelope>,
+    pub consistency: Option<ConsistencyInfo<T>>,
```

Comment:
So, I feel like this should be a property of the connector, rather than of the source. Only Kafka sinks will be able to have a consistency topic, right? And for e.g. file sinks we'd need a "consistency file", so the syntax will be quite different, and inextricably tied up with the connector, it seems. 

(To back up for a second, I think we misdesigned `FORMAT`, and it ought to have been part of the connector too. For example, Kafka sources have key _and_ value formats, while most other connectors only have one format. I think there's some work in progress to try to clean that up.)

Reply:
Maybe:

```sql
CREATE SINK ... INTO
KAFKA BROKER 'blah' 
TOPIC ...
[CONSISTENCY TOPIC ... [FORMAT ...]]
[ENVELOPE ...]
[WITH ...]
```

That reads pretty nicely to me!

Reply:
> (To back up for a second, I think we misdesigned FORMAT, and it ought to have been part of the connector too. For example, Kafka sources have key and value formats, while most other connectors only have one format. I think there's some work in progress to try to clean that up.)

Ah, that makes sense to me! I will switch this around.

Reply:
The only real downside of putting the consistency information on the connector is that it will be present for sources as well. Are we okay with that? Should we have separate sink and source connectors?

Reply:
> Should we have separate sink and source connectors?

Yes, I think so! These two things are proving not to be very symmetric. But I'd say don't fight that fight unless you want to, and if you do, do it in another PR. I think it's 100% fine to have another `Option`al field for now that just gets verified to be `None` in the `CREATE SOURCE` path.